chromadb/segment/impl/manager/local.py

```diff
@@ -150,6 +197,9 @@ def get_segment(self, collection_id: UUID, type: Type[S]) -> S:
             raise ValueError(f"Invalid segment type: {type}")
 
         if scope not in self._segment_cache[collection_id]:
+            memory_limit = self._system.settings.require("chroma_memory_limit_bytes")
+            if type == VectorReader and self._system.settings.require("is_persistent") and memory_limit > 0:
+                self._cleanup_segment(collection_id, memory_limit)
```

Comment:
nit: cleanup_segment could be renamed a bit more explicitly / when we have a proper lru cache abstraction for this